aea/protocols/generator.py

```diff
@@ -161,56 +161,67 @@ def _message_class_str(self) -> str:
         cls_str += '    def __init__(self, message_id: int, target: int, performative: str, contents: Dict, **kwargs):\n'
         cls_str += '        \"\"\"Initialise.\"\"\"\n'
         cls_str += '        super().__init__(message_id=message_id, target=target, performative=performative, contents=contents, **kwargs)\n\n'
-
-        # variables
-        # cls_str += str.format('        self.performatives = {}\n', self._performatives_set())
-        cls_str += str.format('        self.speech_acts = {}\n\n', self._speech_acts_str())
+        cls_str += str.format('        self.speech_acts_definition = {}\n\n', self._speech_acts_str())
         cls_str += '        assert self.check_consistency()\n\n'
 
         cls_str += '    @property\n'
-        cls_str += '    def performatives(self) -> set:\n'
+        cls_str += '    def message_id(self) -> int:\n'
+        cls_str += '        \"\"\"Get the message_id of the message.\"\"\"\n'
+        cls_str += '        assert self.is_set("message_id"), "message_id is not set"\n'
+        cls_str += '        return cast(int, self.get("message_id"))\n\n'
+
+        cls_str += '    @property\n'
+        cls_str += '    def target(self) -> int:\n'
+        cls_str += '        \"\"\"Get the target of the message.\"\"\"\n'
+        cls_str += '        assert self.is_set("target"), "target is not set."\n'
+        cls_str += '        return cast(int, self.get("target"))\n\n'
+
+        cls_str += '    @property\n'
+        cls_str += '    def performative(self) -> str:\n'
+        cls_str += '        \"\"\"Get the performative of the message.\"\"\"\n'
+        cls_str += '        assert self.is_set("performative"), "performative is not set"\n'
+        cls_str += '        return cast(str, self.get("performative"))\n\n'
+
+        cls_str += '    @property\n'
+        cls_str += '    def contents(self) -> Dict:\n'
+        cls_str += '        \"\"\"Get the contents of the message.\"\"\"\n'
+        cls_str += '        assert self.is_set("contents"), "contents is not set"\n'
```

Comment:
We need to dynamically generate a property for each one of the content `names` to expose their `value`. So if the contents = {'price':1, 'size':2} then there need to be properties for `price` and `size`

Reply:
You mean for all contents of all allowed performatives?

What if there are some performatives with similarly named contents? Say "request(resource = r)" and "promise(resource r)".

Reply:
Actually currently, "cfp" and "propose" both have a content named "query" of type "DataModel".

Reply:
Well yes, but we should have a restricting which says that a name must be of the same type for all performatives. I don't see why we would want to support a scenario where `my_var = int` for CFP and `my_var = str` for Propose. The developer should use different variable names for different things. Any given message instantiation has one performative only so if the above condition (unique type per var name across performatives) is satisfied there is no problem. The current implementation means I still have to extract from contents and then cast to a type. The same issue applies in this case that we should not allow the above (`my_var = int` for CFP and `my_var = str` for Propose) and the UX is much worse.

Reply:
And note, `query = DataModel` in more than one performative is completely compatible with the approach we have implemented in the protocols and I am suggesting.

Reply:
Done